### PR TITLE
Add support for amber rst7 files and make jobid 1-based

### LINKED ISSUE #838: Support for amber rst7 system files
I've realized that to run the SAMPLing calculations I'll have to add support for `rst7` files as an alternative to `inpcrd`. This is already supported, but it needs to be allowed in the YAML syntax. I have a local fix ready. I'll push it tomorrow.

Question: do we need to enforce `phase1_path` and `phase2_path` to have the same type of input files (e.g., both phases have to use amber files)? The two phases are completely independent calculations so I'm tempted to remove this restriction. Any objection?

### LINKED ISSUE #839: Make --jobid a 1-based id
To run YANK with a job array, we have implemented the command line options `--jobid` and `--njobs`. Only schedulers usually use 1-based ids while YANK uses 0-based. We should probably use 1-based job ids for YANK too to avoid doing arithmetic in bash in the LSF script.

### PR DESCRIPTION
**Pull request by andrrizzi**
- Fix #838: Support for amber rst7 system files.
- Fix #839: Make jobid 1-based.

After this is merged, I'd make a 0.19.3 release that we can refer to in the SAMPL paper, if there are no objections.

### CODE REVIEW

File: Yank/experiment.py [hunk 1]
```diff
@@ -1657,16 +1659,21 @@ def _is_pipeline_solvent_with_receptor(field, solvent_id, error):
             schema:
                 type: string
                 validator: file_exists
-            dependencies: phase2_path
```
Comment (Lnaden) [inline Yank/experiment.py]: Why did you remove the dependency on `phase2_path` here?

File: Yank/experiment.py [hunk 2]
```diff
@@ -502,7 +502,7 @@ class ExperimentBuilder(object):
         can load it later by using :func:`parse` (default is None).
     job_id : None or int
         If you want to split the experiments among different executions,
-        you can set this to an integer 0 <= job_id <= n_jobs-1, and this
+        you can set this to an integer 1 <= job_id <= n_jobs, and this
```
Comment (Lnaden) [inline Yank/experiment.py]: I don't see it in the yaml docs, but just to make sure: `job_id` is not a parameter which can be YAML specified, correct?

Comment (andrrizzi) [reply Yank/experiment.py]: Correct! It's just a CLI option to make it work with array jobs.

File: Yank/experiment.py (hunk 1 above)
Comment (andrrizzi) [reply Yank/experiment.py]: For the records, we talked about this, and we decided to postpone trying to rework the schema to use `required: yes` and `excludes: [receptor, ligand, solute]` in a future PR.

I'll re-add the `dependencies` solution here before merging!

**Comment by Lnaden** [comment]
I can't approve the codeclimate issue, which seems odd. Can you?

**Comment by andrrizzi** [comment]
Odd. I can't find anything to click to approve either.
